etcdmgmt/etcd-mgmt_test.go

```diff
@@ -33,33 +62,33 @@ func TestWriteETCDPidFile(t *testing.T) {
 	defer heketitests.Patch(&etcdPidFile, "/a/b/c/d/etcd.pid").Restore()
 	tests.Assert(t, writeETCDPidFile(cmd.Process.Pid) != nil)
 	cmd.Process.Kill()
+	cmd.Process.Wait()
 }
 
 func TestIsETCDStartNeeded(t *testing.T) {
 	// check once etcd process is running isETCDStartNeeded returns false
 	os.Remove(etcdPidFile)
-	cmd := exec.Command("etcd")
+	cmd := formETCDCommand()
 	err := cmd.Start()
 	tests.Assert(t, err == nil)
 	err = writeETCDPidFile(cmd.Process.Pid)
 	tests.Assert(t, err == nil)
+	time.Sleep(15 * time.Second)
 	start, _ := isETCDStartNeeded()
 	tests.Assert(t, start == false)
 
-	// check once etcd process is killed isETCDStartNeeded returns true
-	//TODO : the last test fails here, need to investigate why, commenting
-	//out the whole section for now
-	/*var pid int
-	oldPid := cmd.Process.Pid
+	//check once etcd process is killed isETCDStartNeeded returns true
 	err = cmd.Process.Kill()
 	tests.Assert(t, err == nil)
-	start, pid = isETCDStartNeeded()
-	tests.Assert(t, oldPid == pid)
-	tests.Assert(t, start == true)*/
+	_, err = cmd.Process.Wait()
+	tests.Assert(t, err == nil)
+	start, _ = isETCDStartNeeded()
+	tests.Assert(t, start == true)
 
 	// check if the pid file is missing then isETCDStartNeeded returns true
 	os.Remove(etcdPidFile)
 	start, _ = isETCDStartNeeded()
 	tests.Assert(t, start == true)
 	cmd.Process.Kill()
```

Comment:
yeah, it's better to handle these error in one place for all review comments/uncomment code to remove duplicity